File: A2/A2-026/A2-026.cpp
#include <bits/stdc++.h>
#include <string>

int main() {
  int n;
  std::cin >> n;

  std::string name, heaviestName;
  int weight;
  int countOverweight = 0;
  int heaviestWeight = -1;
  const double LIMIT = 15;

  std::set<int> seenWeights;

  for (int i = 0; i < n; ++i) {
    std::cin >> name >> weight;

    if (seenWeights.count(weight)) {
      continue; // Skip repeated weights
    }

    seenWeights.insert(weight);

    if (weight > LIMIT) {
      countOverweight++;
    }

    if (weight > heaviestWeight) {
      heaviestWeight = weight;
      heaviestName = name;
    }
  }

  std::cout << countOverweight << std::endl;
  if (heaviestWeight != -1) {
    std::cout << heaviestName << " " << heaviestWeight << std::endl;
  } else {
    std::cout << "No valid input\n";
  }

  return 0;
}

File: A2/A2-026/A2-026.py
n = int(input())

count_overweight = 0
heaviest_weight = -1
heaviest_name = ''
LIMIT = 15

for _ in range(n):
    name, weight = input().split()
    weight = int(weight)

    if weight > LIMIT:
        count_overweight += 1

    if weight > heaviest_weight:
        heaviest_weight = weight
        heaviest_name = name

print(count_overweight)
print(f"{heaviest_name} {heaviest_weight}")
